policy/release/trusted_artifacts.rego

```diff
@@ -0,0 +1,141 @@
+#
+# METADATA
+# title: Trusted Artifacts
+# description: >-
+#   Trusted Artifacts ensure that only trusted Tasks were used from the
+#   origin of the source code to the produced image. The chain is
+#   constructed by linking specific parameters with results -- ones that
+#   are named `*_ARTIFACT` and have values in the form of
+#   `sha-256:<digest>`. If a chain of same-valued digests can be formed
+#   from the first Task result to the last Task by correlating the
+#   digest values of those and any tasks inbetween.
+#
+#   Trusted Tasks should be producing and using only the files with the
+#   same digest. Given this assurance any code injection should be
+#   detected by observing a modified digest.
+#
+#   This allows addition of Tasks on the Pipeline that do not need to be
+#   trusted, i.e. custom tasks.
+#
+package policy.release.trusted_artifacts
+
+import rego.v1
+
+import data.lib
+import data.lib.tkn
+
+_supported_uris_reg := {"oci:.*@sha256-[0-9a-f]{64}"}
```

Comment:
This needs to be `sha256:` instead of `sha256-`.

(We should probably also have a policy rule that raises an error if some `_ARTIFACT` result/parameter doesn't conform to these?)

Reply:
Perhaps a warning? Though if we reached this point the review on the Task failed already

Reply:
Good point. Maybe let's leave it as is for now.